### PR TITLE
Tribonacci numbers and constant

### PR DESCRIPTION
**Pull request by raoulb**
Add the tribonacci numbers and limit constant.

### CODE REVIEW

**Comment by raoulb** [comment]
I have no idea why this test fails:

```
In [1]: t = (TribonacciConstant + x).evalf()
In [2]: u = TribonacciConstant.evalf() + x

In [3]: t == u
Out[3]: False
```

There seems to be a small difference in some internal format:
```
In [4]: t.as_terms()
Out[4]: ([(1.83928675521416, ((1.8392867552141607, 0.0), (0,), ())), (x, ((1.0, 0.0), (1,), ()))], [x])

In [5]: u.as_terms()
Out[5]: ([(1.83928675521416, ((1.8392867552141612, 0.0), (0,), ())), (x, ((1.0, 0.0), (1,), ()))], [x])
```

How should we fix this?

**Comment by raoulb** [comment]
Did anyone look at this failure?

**Comment by jksuom** [comment]
I'm not sure how this should be fixed, but the reason for the failure is that `.eval()` will automatically [add 4](https://github.com/sympy/sympy/blob/master/sympy/core/evalf.py#L1400) to the (binary) precision using 57 instead of the default 53 when evaluating the constant value `TribonacciConstant.evalf()`. It cannot evaluate `TribonacciConstant + x` as the value of `x` is unknown, and then it falls back to `._eval_evalf()` with default precision.
```
>>> (TribonacciConstant + x).evalf() == TribonacciConstant._eval_evalf(53) + x
True
```

**Comment by raoulb** [comment]
Ok, this explains things at least. For the `GoldenRatio` we have
https://github.com/raoulb/sympy/blob/3f2dbc81188744cb6b7591844a65959962470691/sympy/core/numbers.py#L3578
But the Tribonacci constant is not implemented in the mpmath library. Maybe we should compute is by its explicit formula but using functions for mpmath?

**Comment by raoulb** [comment]
Hmm, still failing :/

File: sympy/core/numbers.py [hunk 1]
```diff
@@ -3623,6 +3623,73 @@ def _sage_(self):
...
+    __slots__ = []
+
+    def _latex(self, printer):
+        return r"\mathrm{TribonacciConstant}"
+
+    def __int__(self):
+        return 2
+
+    def _eval_evalf(self, prec):
+        rv = self._eval_expand_func(function=True)._eval_evalf(prec + 10)
```
Comment (jksuom) [inline sympy/core/numbers.py]: Perhaps it would suffice to set `prec + 4` as in `evalf()`.

**Comment by raoulb** [comment]
Can we merge this now?

**Comment by jksuom** [comment]
This looks fine to me.

**Comment by raoulb** [comment]
Thanks for merging.

**Pull request merged**